api/src/main/java/net/kyori/adventure/text/format/TextColor.java

```diff
@@ -188,6 +188,35 @@
     return this.value() & 0xff;
   }
 
+  /**
+   * Linearly interpolates between {@code this} and {@code that} by {@code t}.
+   *
+   * @param t interpolation value
+   * @param that upper bound
+   * @return the interpolated value
+   * @since 4.3.0
+   */
+  default @NonNull TextColor lerp(final float t, final @NonNull TextColor that) {
```

Comment:
Unsure about this non-`static` method personally.

Reply:
I think the non-static method should be the primary one here -- we own `TextColor`, so it makes sense to put methods directly on it.